File: src/com/timepath/hl2/io/demo/PacketHandler.java
package com.timepath.hl2.io.demo;

import com.timepath.Pair;
import com.timepath.io.BitBuffer;
import java.util.List;

/**
 *
 * @author TimePath
 */
public abstract class PacketHandler {

    boolean read(BitBuffer bb, List<Pair<Object, Object>> l, HL2DEM demo) {
        return false;
    }

}

File: src/main/kotlin/com/timepath/hl2/io/demo/PacketHandler.kt
package com.timepath.hl2.io.demo

import com.timepath.io.BitBuffer

interface PacketHandler {

    public fun read(bb: BitBuffer, l: MutableList<Pair<Any, Any>>, demo: HL2DEM, lengthBits: Int = -1): Boolean = false
}
